### PR TITLE
Fix Workflow plug-in version and Building from source section

### PR DESCRIPTION
**Pull request by ljmotta**
This PR updates the kn_cli_version to 0.21.3 (release in progress), and updates the Building from source section.

Please make sure that your PR meets the following requirements:

- [ ] You have read the [contributions doc](https://github.com/kiegroup/kogito-docs/blob/main/CONTRIBUTING.md)
- [ ] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [ ] Pull Request title contains the target branch if not targeting main: `[0.9.x] KOGITO-XYZ Subject`
- [ ] The nav.adoc file has a link to this guide in the proper category
- [ ] The index.adoc file has a card to this guide in the proper category, with a meaningful description

### CODE REVIEW

**Review by ricardozanini — APPROVED** [review]

**Review by spolti — APPROVED** [review]

File: serverlessworkflow/modules/ROOT/pages/tooling/kn-plugin-workflow-overview.adoc [hunk 1]
```diff
@@ -297,12 +297,14 @@ You can use the `kubectl` command line if you want to use a complex deployment s
 --
 
 [[proc-build-sw-project-source]]
-== Building a workflow project from source
+== Building from source
 
-As an alternative, you can build your workflow project from source.
+As an alternative, you can build the Workflow plug-in from source.
```
Comment (hmanwani-rh) [inline serverlessworkflow/modules/ROOT/pages/tooling/kn-plugin-workflow-overview.adoc:302]:
```suggestion
As an alternative, you can build the workflow plug-in from the source.
```

File: serverlessworkflow/modules/ROOT/pages/tooling/kn-plugin-workflow-overview.adoc [hunk 2]
```diff
@@ -315,30 +317,28 @@ git clone git@github.com:kiegroup/kie-tools.git
...
 [source,shell]
 ----
 cd kie-tools
 pnpm bootstrap
-pnpm -F @kie-tools/kn-plugin-workflow... build:dev
 ----
 --
 
-. Go to the plug-in folder and enter the following command to build the workflow project:
+. In the repository root, enter the following command to build the Workflow plug-in:
```
Comment (hmanwani-rh) [inline serverlessworkflow/modules/ROOT/pages/tooling/kn-plugin-workflow-overview.adoc:331]:
```suggestion
. In the repository root, enter the following command to build the workflow plug-in:
```

File: serverlessworkflow/modules/ROOT/pages/tooling/kn-plugin-workflow-overview.adoc [hunk 3]
```diff
@@ -315,30 +317,28 @@ git clone git@github.com:kiegroup/kie-tools.git
...
 +
 --
-.Build workflow project from source
+.Build Workflow plug-in from source
```
Comment (hmanwani-rh) [inline serverlessworkflow/modules/ROOT/pages/tooling/kn-plugin-workflow-overview.adoc:334]:
```suggestion
.Build workflow plug-in from source
```

File: serverlessworkflow/modules/ROOT/pages/tooling/kn-plugin-workflow-overview.adoc [hunk 4]
```diff
@@ -315,30 +317,28 @@ git clone git@github.com:kiegroup/kie-tools.git
...
 [source,shell]
 ----
-cd packages/kn-plugin-workflow
-go mod tidy
+pnpm -r -F kn-plugin-workflow... build:dev
 ----
 --
 
-You have successfully built your workflow project from source.
+You have successfully built the Workflow plug-in project from source. This _Procedure_ generates an artifact compatible with your OS architecture in the `kie-tools/packages/kn-plugin-workflow/dist` folder.
```
Comment (hmanwani-rh) [inline serverlessworkflow/modules/ROOT/pages/tooling/kn-plugin-workflow-overview.adoc:341]:
```suggestion
You have successfully built the workflow plug-in project from the source. This procedure generates an artifact compatible with your operating system architecture in the `kie-tools/packages/kn-plugin-workflow/dist` folder.
```

**Review by hmanwani-rh — COMMENTED** [review]
@ljmotta Added some suggestions below. Please take a look and let me know if you've any questions.

**Comment by ljmotta** [comment]
@hmanwani-rh Thanks for reviewing. I've made your suggestions.

**Comment by ricardozanini** [comment]
@ljmotta can u please rebase?

**Pull request merged**